### ISSUE TITLE
Usage of node http module blocks usage of next.js

### ISSUE THREAD

**Issue by auchenberg** [stripe/stripe-node#796]
When using Next.js that does isomorphic rendering sometimes React components will be rendered on the client and other times server-side in a Node context.

Steps:
1. Create new next.js app.
1. Modify `pages/index.js` to:
```
import fetch from "isomorphic-unfetch";

var stripe = require("stripe")("<stripe key>");

function HomePage({ invoice }) {
  return <div>{invoice.id}</div>;
}

HomePage.getInitialProps = async () => {
  const invoice = await stripe.invoices.retrieve("<invoice id>");

  return { invoice: invoice };
};

export default HomePage;
```
Actual:

<img width="1055" alt="Screen Shot 2020-02-05 at 7 33 18 PM" src="https://user-images.githubusercontent.com/173559/73903605-62c5ca80-484e-11ea-8df6-b4ae2a1cb147.png">

```
TypeError: __webpack_require__(...).createServer is not a function
(anonymous function)
./node_modules/stripe/lib/stripe.js:11
   8 | const DEFAULT_API_VERSION = null;
   9 | 
  10 | // Use node's default timeout:
> 11 | const DEFAULT_TIMEOUT = require('http').createServer().timeout;
  12 | 
  13 | Stripe.PACKAGE_VERSION = require('../package.json').version;
  14 | 
```

Expected:
`stripe-node` should detect if Node context and `http` module is available and it's safe load.

**Comment by jonchurch** [comment]
This doesn't relate directly to your issue but I wanted to mention that the secret API keys used for this library are intended to never be exposed on the client. See the [Authentication Docs](https://stripe.com/docs/api/authentication). 

I would recommend creating an [API route in Next.js](https://nextjs.org/docs/api-routes/introduction) to make this call to the Stripe API to avoid exposing your secret key on the client, as well as fix the issue of not being able to use this library on the client side. Your frontend could then make an API call to that API route and still get the data it needs.

You could implement the functionality of this library yourself using `fetch` to manually make the Stripe API call from the client, but again that is recommended against by the Stripe docs because it would force you to expose your secret key in the frontend. 

I hope this helped ❤️

**Comment by auchenberg** [comment]
Let me provide another example that shows the problem. Here the Stripe library is never initialized client-side, but client-side rendering still fails because the `stripe` library requires `http` to be present even at parse/runtime. Not at initialization, which only happens in the server-side case. 

```
var stripe = require("stripe")

function Page({ invoice }) {

  return (
    <Layout>

    </Layout>
  );
}

Page.getInitialProps = async ({ query }) => {

  if (!process.browser) {
    // Only execute this code in a server-side context.
    let stripeClient = stripe(process.env.STRIPE_SECRET);
    let invoiceId = query.id;

    const invoice = await stripeClient.invoices.retrieve(invoiceId);

    return { invoice: invoice };
  }

};

export default Page;
```

This can be mitigated by moving the `require` statement into the server-side only context, but this seems like a workaround.

```

function Page({ invoice }) {

  return (
    <Layout>

    </Layout>
  );
}

Page.getInitialProps = async ({ query }) => {

  if (!process.browser) {
    var stripe = require("stripe")

    // Only execute this code in a server-side context.
    let stripeClient = stripe(process.env.STRIPE_SECRET);
    let invoiceId = query.id;

    const invoice = await stripeClient.invoices.retrieve(invoiceId);

    return { invoice: invoice };
  }

};

export default Page;
```

**Comment by auchenberg** [comment]
This was fixed in https://github.com/stripe/stripe-node/commit/85e11b60665f66cd21d19d4b9c5fc6f1639677f0, and have been validated to resolve the problem. Closing.

**Comment by jonchurch** [comment]
Ahh it seems I misunderstood what you were attempting to do anyways, glad it was resolved.

I don't want to sound pedantic, but still I'd strongly suggest putting this request into an API route. I don't fully understand how Next.js bundles it's code, but I believe `getInitialProps` can still be executed client side, leading me to believe it gets bundled up and sent to the browser. If you are sending your secret key to the client, that's a problem.

I also found [this article](https://arunoda.me/blog/ssr-and-server-only-modules) about excluding certain modules from your webpack build which I thought was interesting. 

However, when it comes to an API key that unless properly restricted gives access to your entire Stripe account, I would urge you to make sure that logic only runs on your server.
